lib/ddtrace/contrib/resque/configuration/settings.rb

```diff
@@ -23,8 +23,16 @@ class Settings < Contrib::Configuration::Settings
           end
 
           option :service_name, default: Ext::SERVICE_NAME
+
+          # TODO: 1.0: When moving to auto patching all workers by default
+          # we should remove this setting, as it will be a no-op.
           option :workers, default: []
           option :error_handler, default: Datadog::Tracer::DEFAULT_ON_ERROR
+
+          # TODO: 1.0: Automatic patching should be the default behavior.
+          # We should not provide this option anymore when making it the default,
+          # as our integrations should always instrument all possible scenarios when feasible.
+          option :auto_instrument, default: false
```

Comment:
That's too smart for my brain.

Reply:
I was thinking about this, and the main issue with `use_v1_defaults` is that what `use_v1_defaults` means today will likely change tomorrow, when we decide to schedule the next breaking change for 1.0.

For example, if we had introduced `use_v1_defaults` in the previous release, this release would change how Redis workers are configured, effectively being a breaking change for `use_v1_defaults` users.
It would effectively mean that `use_v1_defaults` == "every release can be a breaking release".

Reply:
For this PR though, I think maybe the best thing to do is, for users that have `c.use resque`, but no `workers` setting, to effectively become `auto_instrument`.
Maybe change the semantics: remove `auto_instrument` option, and allow empty `workers` to be "auto-instrument".

What do you think? There's no right or wrong here: users with an explicit `worker: nil` will likely see instrumentation that they didn't expect, but a `nil` value is not even supported here.

Reply:
> It would effectively mean that use_v1_defaults == "every release can be a breaking release".

Yes, but you'd be signing up for those semantics for `use_v1_defaults` (e.g. that the defaults may change, as they adopt the v1 conventions), so I don't see that as a big issue. (I may be underestimating how often we run into issues, but my expectation is that if we consider something as the new default, that we're reasonably confident that it won't break)

> What do you think? There's no right or wrong here: users with an explicit worker: nil will likely see instrumentation that they didn't expect, but a nil value is not even supported here.

Seems pretty reasonable, and it does the right thing for new customers, I like it!